src/main.cc

```diff
@@ -884,6 +884,15 @@ mainReconfigureStart(void)
              false);
 }
 
+/// error message to log when Configuration::Parse() fails
+static SBuf
+ConfigurationFailureMessage()
+{
+    const auto context = reconfiguring ? "reconfiguration" : "configuration";
+    return ToSBuf(context, " failure: ", CurrentException,
+                  Debug::Extra, "advice: Run 'squid -k parse' and check for ERRORs.");
```

Comment:
This advice used to be conditional on opt_parse_cfg_only. I am happy to restore that, but it feels that if we are going to be in business of giving advice, we should give advice every time that advice may be relevant, including FATAL crashes outside of `-k parse` tests.

Reply:
This advice is given even if we start `squid -k parse` itself.

Reply:
Ha! Good point. Fixed in branch commit 811191f9. 

It is not easy to trigger that official code, but, in startup context, official code only prints that advice in `squid -k parse` mode, where it is useless. This PR now fixes that bug as well.